### PR TITLE
Also run automerge check after pull review

### LINKED ISSUE #24445: Automerge does not work
### Description

I added next rules for merging PR
* All statuses of commit must be successful before merge
* PR must have one approve review 

### Gitea Version

1.19.2

### Can you reproduce the bug on the Gitea demo site?

I don't know (I can't check it)

### Log Gist

_No response_

### Screenshots

image
image

### Git Version

git version 2.25.1

### Operating System

ubuntu 20.04 (Linux beloved-pipefish 5.4.0-125-generic #141-Ubuntu SMP Wed Aug 10 13:42:03 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux)

### How are you running Gitea?

I run Gitea from prepared binary files 

### Database

SQLite

### LINKED ISSUE #30658: Can not create merge commit when checks succeed
### Description

image

reproduce:
user A: site admin
user B: normal user
- user A: create an individual repo
- user A: add user B as a contributor with write permission
- user A: push some code to this repo with an action workflow
- user A: add a branch rule to main with 
  push whitelisted (only user B), required approvals: 1, enable status check
- user B: create a new branch
- user B: create PR
- user B: push a commit to this new branch
- wait until required jobs finished with success status
  updated: important, if jobs are in pending, you will not get any error
- user B: click that blue button

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by 6543**
Close #24445
Fix #30658

### CODE REVIEW

**Comment by rombert** [comment]
Is this related to https://github.com/go-gitea/gitea/issues/24445 ?

**Comment by 6543** [comment]
most likely - I'll have to create a serous pull, this was more a reminder to let me finish it

**Comment by lunny** [comment]
Automerge feature needs a redesign like Github did.

**Review by wxiaoguang — APPROVED** [review]

**Comment by 6543** [comment]
hmm it affects the integration tests ... I need to look if we depend on the broken behavior in our tests of if tests are not isolated enough

**Comment by yp05327** [comment]
Not know what happened, but auto merge not work, and got this error:
image

**Comment by yp05327** [comment]
@lunny 
It seems that still some problems here.
This is the reproduce(operation history):
image
ps: repo owner is the site admin (giteaadmin)

It seems that after testuser enable auto merge, and giteaadmin approved, the auto merge will be canceled.

File: services/automerge/automerge.go
Comment (yp05327) [inline services/automerge/automerge.go:129]: Do we really need to do this here?
As I mentioned in https://github.com/go-gitea/gitea/pull/25741#issuecomment-2071515137
`HeadCommitID` is already be loaded, then you will read it again here.

Comment (lunny) [reply services/automerge/automerge.go:129]: It's different between `HeadCommitID` and this `commitID`.

Comment (yp05327) [reply services/automerge/automerge.go:129]: I see. It is BaseRepo's commit ID.

Comment (yp05327) [reply services/automerge/automerge.go:129]: Can you rename the variable name to `baseGitRepo`?

**Comment by yp05327** [comment]
@lunny 
Another big problem, Gitea server will freeze for a long time after the following steps:
image

reproduce:
shut down the runner
user B:  create a new branch and PR, and enable auto merge
user A: approve it, then we will wait for the CI
start your runner, let the CI finish, then it should be merged as expected, but no, Gitea server will freeze for a long time.

Not sure what happened, I have no time to look into it.

**Comment by lunny** [comment]
replaced by #30780

**Pull request closed without merging**
